src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -1235,8 +1236,18 @@ private async Task HandleBreakModeEvent(ResultEventArgs results, BreakRequest br
                     }
                 }
             }
-            else if (reason == "end-stepping-range" || reason == "function-finished")
+            // step over/into
+            else if (reason == "end-stepping-range")
+                _callback.OnStepComplete(thread);
+            // step out
+            else if (reason == "function-finished")
             {
+                string resultVar = results.Results.TryFindString("gdb-result-var"); // a gdb value history var like "$1"
+                if (!string.IsNullOrEmpty(resultVar))
+                {
+                    ReturnValue = new VariableInformation("Function return value", resultVar, cxt, Engine, (AD7Thread)thread.Client, isParameter: false);
```

Comment:
To keep in sync with what we have in VS, it should be `$ReturnValue`.

```suggestion
ReturnValue = new VariableInformation("$ReturnValue", resultVar, cxt, Engine, (AD7Thread)thread.Client, isParameter: false);
```

Reply:
Nu11